### PR TITLE
Large avatars should instantly shrink when entering small height domains

### PR DESCRIPTION
**Pull request by hyperlogic**
Also, reduce the fudge factor for safeLanding from an additional 50% of avatar's height to 25%.  This should prevent large avatars that enter the start domain from being lifted to the 3rd floor/gray area, or being pushed downward below the domain.

### CODE REVIEW

**Comment by hyperlogic** [comment]
https://highfidelity.fogbugz.com/f/cases/14092/Avatar-becomes-stuck-in-the-wall-Kicked-outside-room-when-navigating-to-Start-domain

**Comment by hyperlogic** [comment]
QA Test plan:

1. Launch Interface.exe (try HMD mode and desktop mode)
2. Make sure collision is enabled (it's normally is set to enabled by default)
3. Navigate to the archilogic domain
4. Move far away from the house model, otherwise your frame rate will be very poor when you increase your avatar size.  (This is a known issue, separate from the issue that this PR addresses)
4. Increase your avatar size, 100x normal.  Keep pressing the + key, or issue the following Console command `MyAvatar.scale = 100`
5. Navigate to the Start domain

Expected result:
User is placed in the Start room, facing windows and the greeter kiosks. The User should not end up in the ceiling (gray room) or below the floor.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10440/HighFidelity-Beta-PR13332-608575c.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10440/HighFidelity-Beta-PR13332-608575c.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10440/HighFidelity-Beta-Interface-PR13332-608575c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10440/HighFidelity-Beta-PR13332-608575c.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10440/HighFidelity-Beta-Interface-PR13332-608575c.exe

**Comment by QliCknox** [comment]
Currently Testing

**Comment by QliCknox** [comment]
Stopped Testing

File: interface/src/avatar/MyAvatar.cpp [hunk 1]
```diff
@@ -2423,6 +2428,8 @@ void MyAvatar::restrictScaleFromDomainSettings(const QJsonObject& domainSettings
     setModelScale(_targetScale);
     rebuildCollisionShape();
     settings.endGroup();
+
+    _haveReceivedHeightLimitsFromDomain = true;
```
Comment (AndrewMeadows) [inline interface/src/avatar/MyAvatar.cpp]: This is the only place where `_haveReceivedHeightLimitsFromDomain` is set `true`, but `MyAvatar::restrictScaleFromDomainSettings()` is only ever called on the `DomainHandler::settingsReceived()` signlal. I wonder: Does this PR breaks physics for **serverless** sessions?

**Comment by qligmaloney** [comment]
Currently testing

**Comment by hyperlogic** [comment]
Note: If archilogic is down, any domain that has the default avatar height limit, can be substituted for archilogic.

**Comment by qligmaloney** [comment]
PASS @conklin94122 Tested using locally hosted domain and Eschatology as archilogic and start domains were unavailable on PR build. User never entered domain anywhere besides expected path. User's height was changed to maximum height allowed in domain upon entering.

**Comment by AndrewMeadows** [comment]
I removed the **QA_APPROVED** label.

The **test plan** was incomplete.  The tests need to also be run in a **serverless** environment.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10505/HighFidelity-Beta-PR13332-b837904.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10505/HighFidelity-Beta-PR13332-b837904.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10505/HighFidelity-Beta-Interface-PR13332-b837904.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10505/HighFidelity-Beta-PR13332-b837904.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10505/HighFidelity-Beta-Interface-PR13332-b837904.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10524/HighFidelity-Beta-PR13332-f773aef.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10524/HighFidelity-Beta-PR13332-f773aef.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10524/HighFidelity-Beta-Interface-PR13332-f773aef.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10524/HighFidelity-Beta-PR13332-f773aef.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10524/HighFidelity-Beta-Interface-PR13332-f773aef.exe

**Pull request merged**